plugin/src/main/kotlin/com/vanniktech/maven/publish/sonatype/SonatypeRepositoryBuildService.kt

```diff
@@ -36,19 +40,43 @@ internal abstract class SonatypeRepositoryBuildService : BuildService<SonatypeRe
   // indicates whether we already closed a staging repository to avoid doing it more than once in a build
   var repositoryClosed: Boolean = false
 
+  var buildHasFailure: Boolean = false
+
+  override fun onFinish(event: FinishEvent) {
+    if (event.result is TaskFailureResult) {
+      buildHasFailure = true
+    }
+  }
+
+  override fun close() {
+    if (buildHasFailure) {
+      return
+    }
+
+    val stagingRepositoryId = this.stagingRepositoryId
+    if (stagingRepositoryId != null) {
+      nexus.closeStagingRepository(stagingRepositoryId)
+      if (parameters.automaticRelease.get()) {
+        nexus.releaseStagingRepository(stagingRepositoryId)
```

Comment:
Isn't this a change from what we had before when executing `closeAndReleaseRepository`. With 0.21.0 it always does close and release. And now when upgrading to this version it only does `close`. Why don't we auto release by default?

Reply:
This is how it compares:

Running `publish`:
0.21.0: implictly createst staging repo, uploads
now - flag off (default): creates staging repo, uploads, closes it
now - flag on: creates staging repo, uploads, closes it, releases it

Running `closeAndReleaseRepository` when repo is open
0.21.0: finds repo, if there is just one closes and releases it
now: finds repo, if there is just one closes and releases it

Running `closeAndReleaseRepository` when repo is closed
0.21.0: finds repo, fails
now: finds repo, if there is just one releases it


So `closeAndReleaseRepository` generally behaves like before and if you had `publish` and `closeAndReleaseRepository` running on ci (or you were manually executing them) the end result is the same. The automatic release flag gives you the option to let publish also release it so that you don't need to run `closeAndReleaseRepository` at all. I thought making the automatic release opt in is a bit safer as it's not something you can reverse.

Reply:
Aaah okay. Now i understand it, thanks!